### DISCUSSION TITLE
VPN tunnel same subnets LAN

### DISCUSSION THREAD

**Discussion by cosmicDustOfLightLength** [strongswan/strongswan#1617] [Q&A]
How to combine it and what should the ipsec.conf configuration look like?

Devices A can ping B and C all three can see and ping each other. Being on the same subnet. I don't find such a configuration in the examples and I started to wonder if such a configuration is possible.

And what flags should strongswan have? ./configure compiles(make) it manually.

![obraz](https://user-images.githubusercontent.com/71566296/226733153-c18b7540-9f6a-4b34-ab17-72c8e5b3df24.png)

**Comment by tobiasbrunner** [comment]
> I don't find such a configuration in the examples and I started to wonder if such a configuration is possible.

There is the [ikev2/net2net-same-nets](https://strongswan.org/testing/testresults/ikev2/net2net-same-nets/index.html) test scenario, which uses `NETMAP` (see the `iptables-save` output on sun).

**Comment by cosmicDustOfLightLength** [comment]


**Reply by tobiasbrunner** [reply]
You have clearly not applied the referenced configuration. For instance, you configured virtual IPs (`rightsourceip`) which makes no sense for a site-to-site connection; and you configured the same subnets on both sides (`leftsubnet`), which is precisely what you wanted to avoid (at least as far as I understood, please clarify if that's not what you wanted).

> The IKEv2 client [A] does not use strongswan. It has some IKEv2/ipsec implementation.

That's fine, as long as it does not insist on negotiating a tunnel with the same subnet on both ends. As you can see in the test scenario, only sun requires special firewall rules, the SA is a regular net-to-net one (although with the virtual subnet `10.4.0.0/14` on one end).

**Reply by cosmicDustOfLightLength** [reply]
Thank you in advance for your help.

I wanted the same subnet 192.168.3.0/24 on both sides of the tunnel

So that the connected client [A] is on the same LAN subnet as computer [B] and [C]
So that everyone can "ping" each other using IP addresses from the local subnet 192.168.3.0/24

So the same subnet on both sides of the tunnel is not a good idea and you have to make two different subnets mandatory?

I developed the configuration by trial and error in ipsec.conf, in the example the configuration in the swanctl.conf file is written differently.
If I delete the content of ipsec.conf. I will enter this configuration into swanctl.conf in the notation that swanctl.conf prefers, of course.
Will strongSwan automatically switch to using this configuration?

I use strongSwan through systemd restarting the service on configuration changes.

**Reply by tobiasbrunner** [reply]
> I wanted the same subnet 192.168.3.0/24 on both sides of the tunnel
> 
> So that the connected client [A] is on the same LAN subnet as computer [B] and [C] So that everyone can "ping" each other using IP addresses from the local subnet 192.168.3.0/24

That's not really possible. You might want to learn about IP networking in general, and routing and ARP in particular.

**Reply by cosmicDustOfLightLength** [reply]
Thx is now clear for me!
